Middleware deduping

patrikholcak opened the pull request:
## About

This might be a controversial (and breaking) change, but instead of overriding middlewares, I think the step middleware should always win. Feel free to close this PR or message me to make changes.

Take this example:

```
const defaultStepOptions = {
  floatingUIOptions: {
    middleware: [offset({ crossAxis: 32 })]
  }
};

const stepOptions = {
  floatingUIOptions: {
    middleware: [offset({ crossAxis: -32 })]
  }
}
```

To override `defaultStepOptions` you need equal negative spacing and sync these two numbers every time the default changes (or have a constant)

With this PR:

```
const defaultStepOptions = {
  floatingUIOptions: {
    middleware: [offset({ crossAxis: 32 })]
  }
};

const stepOptions = {
  floatingUIOptions: {
    middleware: [offset({ crossAxis: 0 })]
  }
}
```

The step offset middleware overrides the default step middleware

chuckcarpenter commented:
@patrikholcak we have been talking about this recently and I recently released a change that changes the ordering, so that a dup added in would run last, but that still could have issues. This would be onus on the user to make sure they still have sane defaults when going down this path, but does give the most control.

chuckcarpenter commented:
@patrikholcak docs failed to build due to this error

<img width="1248" alt="image" src="https://github.com/user-attachments/assets/762905bb-2870-4ad0-854c-9f71b085c67a">

patrikholcak commented:
Rebased and updated. I also updated a step test which was failing due to the changes in [dfb7d74](https://github.com/shipshapecode/shepherd/pull/3012/commits/dfb7d743d458e507f1d50b10972c62d1b4ab49ee). I’m not sure the test is relevant now, can probably be removed altogether :shrug:
